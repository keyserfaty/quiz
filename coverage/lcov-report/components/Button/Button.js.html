<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Button/Button.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Button</a> Button.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity, Text, ActivityIndicator } from 'react-native'
&nbsp;
import styles from './statics/styles'
&nbsp;
const Button = props =&gt; {
  const { onPress, children, primary, outline, disabled, style, status } = props
&nbsp;
  const spinnerColor = {
    outline: 'rgba(116, 255, 198, 1)',
    primary: 'white'
  }
&nbsp;
  return (
    &lt;TouchableOpacity 
      disabled={disabled}
      onPress={onPress}
      style={[
        styles.button, 
        primary &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >styles.primary, </span>
        disabled &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >styles.disabled, </span>
        outline &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >styles.outline, </span>
        style
      ]}&gt;
      {status === 'pending' &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >&lt;ActivityIndicator size="small" style={styles.spinner} color={spinnerColor[primary ? 'primary' : 'outline']} /&gt; </span>}
      &lt;Text style={[
        styles.text,
        primary &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >styles.textPrimary,</span>
        outline &amp;&amp; <span class="branch-1 cbranch-no" title="branch not covered" >styles.textOutline</span>
      ]}&gt;{children.toUpperCase()}&lt;/Text&gt;
    &lt;/TouchableOpacity&gt;
  )
}
&nbsp;
Button.propTypes = {
  onPress: PropTypes.func,
  children: PropTypes.any,
  status: PropTypes.string,
  primary: PropTypes.bool,
  outline: PropTypes.bool,
  disabled: PropTypes.bool,
  style: PropTypes.object
&nbsp;
}
&nbsp;
export default Button
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 20:22:18 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
