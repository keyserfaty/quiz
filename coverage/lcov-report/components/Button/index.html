<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Button</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/Button
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/22</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">10.71% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/22</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Button.js"><a href="Button.js.html">Button.js</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="18" class="abs medium">9/18</td>
	<td data-value="21.43" class="pct low">21.43%</td>
	<td data-value="14" class="abs low">3/14</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="18" class="abs medium">9/18</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 06:19:18 GMT-0500 (EST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
