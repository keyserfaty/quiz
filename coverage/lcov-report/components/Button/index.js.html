<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Button/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/Button</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.43% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/18</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react'
import propTypes from 'prop-types'
import { isNil } from 'ramda'
&nbsp;
import Button from './Button'
&nbsp;
class ButtonWithSpinnerContainer extends React.Component {
  constructor (props) {
    super(props)
&nbsp;
    this.state = {
      status: 'init',
      disabled: false
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidMount () {
    const { status, disabled } = <span class="cstat-no" title="statement not covered" >this.props</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      status: !isNil(status) ? status : 'init',
      disabled: !isNil(disabled) ? disabled : false
    })
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidUpdate (oldProps) {
    const { status, disabled } = <span class="cstat-no" title="statement not covered" >this.props</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (this.props.rerender !== oldProps.rerender) {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ status: 'init' })</span>
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (disabled !== oldProps.disabled) {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ disabled })</span>
    }
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (status !== oldProps.status) {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({ status })</span>
    }
  }
&nbsp;
  onPress = () =&gt; {
    if (!this.props.noSpin) {
      this.setState({
        status: 'pending',
        disabled: true
      })
    }
&nbsp;
    <span class="missing-if-branch" title="else path not taken" >E</span>if (this.props.onPress) {
      this.props.onPress()
    }
  }
&nbsp;
  render () {
    const { status, disabled } = this.state
&nbsp;
    return (
      &lt;Button
        {...this.props}
        disabled={disabled}
        status={status}
        onPress={this.onPress}
      /&gt;
    )
  }
}
&nbsp;
ButtonWithSpinnerContainer.propTypes = {
  onPress: propTypes.func.isRequired
}
&nbsp;
export default ButtonWithSpinnerContainer</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Nov 11 2018 06:29:52 GMT-0500 (EST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
