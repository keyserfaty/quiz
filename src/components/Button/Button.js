import React from 'react'
import { TouchableOpacity, Text, ActivityIndicator } from 'react-native'

import styles from './statics/styles'

const Button = props => {
  const { onPress, children, primary, outline, disabled, style, status } = props

  const spinnerColor = {
    primary: 'rgba(55,175,255,0.87)',
    outline: 'rgba(255, 255, 255, 1)'
  }

  return (
    <TouchableOpacity 
      disabled={disabled}
      onPress={onPress}
      style={[
        styles.button, 
        primary && styles.primary, 
        disabled && styles.disabled, 
        outline && styles.outline, 
        style
      ]}>
      {status === 'pending' && <ActivityIndicator size="small" style={styles.spinner} color={spinnerColor[primary ? 'primary' : 'outline']} /> }
      <Text style={[
        styles.text,
        primary && styles.textPrimary,
        outline && styles.textOutline
      ]}>{children.toUpperCase()}</Text>
    </TouchableOpacity>
  )
}

export default Button
