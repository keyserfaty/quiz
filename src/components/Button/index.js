import React from 'react'
import propTypes from 'prop-types'

import Button from './Button'

class ButtonWithSpinnerContainer extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      status: 'init',
      disabled: false
    }
  }

  componentDidMount () {
    this.setState({
      status: this.props.status || 'init',
      disabled: this.props.disabled || false
    })
  }

  componentDidUpdate (oldProps) {
    if (this.props.rerender !== oldProps.rerender) {
      this.setState({ status: 'init' })
    }

    if (this.props.disabled !== oldProps.disabled) {
      this.setState({ disabled: this.props.disabled })
    }
  }

  onPress = () => {
    if (!this.props.noSpin) {
      this.setState({
        status: 'pending',
        disabled: true
      })
    }

    if (this.props.onPress) {
      this.props.onPress()
    }
  }

  render () {
    const { status, disabled } = this.state

    return (
      <Button
        {...this.props}
        disabled={disabled}
        status={status}
        onPress={this.onPress}
      />
    )
  }
}

ButtonWithSpinnerContainer.propTypes = {
  onPress: propTypes.func.isRequired
}

export default ButtonWithSpinnerContainer