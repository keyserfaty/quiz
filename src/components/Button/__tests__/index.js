import React from 'react'

import Button from '../index'

import Shallow from 'react-test-renderer/shallow'

const renderer = new Shallow()

test('Should render', () => {
  let component = renderer.render(<Button onPress={() => {}} />)

  expect(component).toMatchSnapshot('initial render')
})

test('Should keep status on press if noSpin is true', () => {
  let component = renderer.render(<Button noSpin={true} onPress={() => {}} />)
  component = renderer.getRenderOutput()

  expect(component).toMatchSnapshot('initial render')

  component.props.onPress()
  component = renderer.getRenderOutput()
  
  expect(component).toMatchSnapshot('status should be init')
})

test('Should change status on press', () => {
  let component = renderer.render(<Button onPress={() => {}} />)

  expect(component).toMatchSnapshot('initial render')

  component.props.onPress()
  component = renderer.getRenderOutput()
  
  expect(component).toMatchSnapshot('status should be pending')
})
