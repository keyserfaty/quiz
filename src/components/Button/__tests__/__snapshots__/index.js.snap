// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should change status on press: initial render 1`] = `
<Button
  disabled={false}
  onPress={[Function]}
  status="init"
/>
`;

exports[`Should change status on press: status should be pending 1`] = `
<Button
  disabled={true}
  onPress={[Function]}
  status="pending"
/>
`;

exports[`Should keep status on press if noSpin is true: initial render 1`] = `
<Button
  disabled={false}
  noSpin={true}
  onPress={[Function]}
  status="init"
/>
`;

exports[`Should keep status on press if noSpin is true: status should be init 1`] = `
<Button
  disabled={false}
  noSpin={true}
  onPress={[Function]}
  status="init"
/>
`;

exports[`Should render: initial render 1`] = `
<Button
  disabled={false}
  onPress={[Function]}
  status="init"
/>
`;
